Yona/Yona/MeDashBoard/YonaUserProfileViewController.swift

```diff
@@ -317,101 +227,160 @@ class YonaUserProfileViewController: UIViewController, UITableViewDelegate, UITa
-                                                            kCTFontAttributeName: UIFont(name: "SFUIDisplay-Bold", size: 14)!] as? [NSAttributedStringKey : Any]
-    }
-    
-    func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
-        picker.dismiss( animated: true, completion: nil)
-        UINavigationBar.appearance().tintColor = UIColor.yiWhiteColor()
-        UINavigationBar.appearance().titleTextAttributes = [NSAttributedStringKey.foregroundColor : UIColor.white,
-                                                            NSAttributedStringKey.font: UIFont(name: "SFUIDisplay-Bold", size: 14)!]
-        UINavigationBar.appearance().barTintColor = UIColor.yiWhiteColor()
-        
     }
     
     func resizeImage(_ image: UIImage, targetSize: CGSize) -> UIImage? {
         let size = image.size
-        
         let widthRatio  = targetSize.width  / size.width
         let heightRatio = targetSize.height / size.height
-        
         // Figure out what our orientation is, and use that to form the rectangle
         var newSize: CGSize
         if(widthRatio > heightRatio) {
             newSize = CGSize(width: size.width * heightRatio, height: size.height * heightRatio)
         } else {
             newSize = CGSize(width: size.width * widthRatio,  height: size.height * widthRatio)
         }
-        
         // This is the rect that we've calculated out and this is what is actually used below
         let rect = CGRect(x: 0, y: 0, width: newSize.width, height: newSize.height)
-        
         // Actually do the resizing to the rect using the ImageContext stuff
         UIGraphicsBeginImageContextWithOptions(newSize, false, 1.0)
         image.draw(in: rect)
         let newImage = UIGraphicsGetImageFromCurrentImageContext()
         UIGraphicsEndImageContext()
-        
         return newImage
     }
-}
+ }
+ 
+ //MARK: YonaUserHeaderTabProtocol
+ extension YonaUserProfileViewController: YonaUserHeaderTabProtocol {
+    func didSelectProfileTab() {
+        isShowingProfile = true
+        if let itmes = rightSideButtonItems {
+            self.navigationItem.rightBarButtonItems = itmes
+        }
+        tableView.reloadData()
+    }
+    
+    func didSelectBadgesTab() {
+        isShowingProfile = false
+        rightSideButtonItems = self.navigationItem.rightBarButtonItems
+        self.navigationItem.rightBarButtonItems = nil
+        tableView.reloadData()
+    }
+    
+    func didAskToAddProfileImage() {
+        chooseImage(self)
+    }
+ }
+ 
+ //MARK: UITableViewDelegate and UITableViewDataSource
+ extension YonaUserProfileViewController: UITableViewDelegate, UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if section == 0 {
+            return 1
```

Comment:
Same here (but for 1 instead of 2 :) ) and below

Reply:
comment added.